js/how-does-javascript-work/how-does-javascript-work.md

```diff
@@ -0,0 +1,149 @@
+Hello, 👋 this is Shivam Verma, this is my first blog and in this, I will help you to understand the working of JavaScript. Let's Begin!!
+
+#### Ever wondered how JavaScript works under the hood?     
+
+Well by the end of this blog you will have a clear understanding of that by knowing two important concepts that make it possible
+1. **Execution Context**
+1. **Call Stack**          
+
+### 1. Execution Context
+
+Before diving deep into the working of `JavaScript` remember one thing
+> "Everything in JavaScript happens inside an `Execution Context`" 
+
+`Execution Context(EC)` can be classified into two main types:
+- Global Execution Context(GEC)
+- Functional Execution Context(FEC)
+
+I will specify when which `EC` is created, other than that we will refer to it commonly as `Execution Context`
+
+So the `Execution Context` is divided into 2 parts
+- **Memory Component**
+- **Code Component** 
+ 
+The **Memory Component** also known as the **Variable Environment** allocates memory and stores variables and functions as *key: value* pairs.
+
+The **Code Component** also known as the **Thread of Execution** runs through the JavaScript program and executes line by line.
+
+Now If I may present the `Execution Context` visually as a diagram it can be shown as:
+
+![Group 6.png](https://cdn.hashnode.com/res/hashnode/image/upload/v1611776579826/9DOXVBrgh.png)
+
+Another line to keep in mind for working of JavaScript
+> " JavaScript is **Synchronous Single-Threaded** language "
+
+This means that JavaScript can only execute one command at a time in a specific order. It can only go to the next line only if the current line is finished executing.
+
+I know you might be thinking that, What about asynchronous programming in JavaScript? Is there no asynchronous programming in JavaScript? Did I get fooled by that Indian guy on YouTube?
```

Comment:
Can you rephrase it so you don't have to say "asynchronous programming in JavaScript" twice?  

Also, I think the Indian guy on youtube is a little inappropriate 😅
How about we try something else?